libraries/AP_Arming/AP_Arming.cpp

```diff
@@ -577,10 +577,13 @@ bool AP_Arming::rc_arm_checks(AP_Arming::Method method)
             }
         }
 
-        if (rc().arming_check_throttle()) {
-            const RC_Channel *c = rc().channel(rcmap->throttle() - 1);
+        // only check throttle inputs if not skipped and not flying
+        if (rc().arming_check_throttle() && !AP_Notify::flags.flying) {
+            RC_Channel *c = rc().channel(rcmap->throttle() - 1);
+            // this assumes throttle range is [0, 100] or [-100, 100] when reversible
+            constexpr int16_t max_limit = 5;
             if (c != nullptr) {
-                if (c->get_control_in() != 0) {
+                if (abs(c->get_control_in()) > max_limit) {
```

Comment:
This is an unacceptable range for the purpose of this check. It *has* to be neutral or it's a safety risk. Why do you want a larger range?

Reply:
@WickedShell Thanks for reviewing. The 5% limit and the is_flying check were suggested by @tridge 
The 5% is to avoid problems with trim and the is_flying check to avoid problems with in-air rearming.
I did say *potential* safety enhancement :)
I should note that this PR resulted from discussion of #14520, where having a VTOL forward motor spinning on arming was considered a safety issue.  
I assume you are currently setting the ARMING_CHECK option and relying on the hard zero limit?

Reply:
Tridge was concerned about non-arms support calles with too strict a limit

Reply:
Yeah, I'm setting it now and relying on the hard 0 limit. I can cope with the bit meaning flipping, but anything non 0 is a real problem for me...

Can you explain how the 5% fixes allow in air rearming?

I admit I had not looked closely enough to see the is_flying bit of logic there. Honestly I'd prefer to not have that creep into this check as well. I guess that could be a separate default check, but as I've fooled is_flying numerous times before I'd really rather know that my arming stick checks are strictly run, and always looking at 0% throttle input....

Reply:
> Tridge was concerned about non-arms support calls with too strict a limit

That's what the purpose of a deadzone is. We are checking that the input is still within the deadzone (or is so barely outside of it it maps to 0). This change would allow me to arm with 5% throttle output and immediately hurt myself or an operator, which is not good.

Also note that this doesn't make sense because it's making throttle more different then all the other stick arming checks.

Reply:
Also a futher note is that you can't actually assume throttle is 0-100. On copter it's 0-1000. Comparing to 0 is fine because that's neutral, but the value of 5 has a totally different meaning between vehicles, and this is a common arming check.

Reply:
OK, so we need to address at least 2 problems:
1) The+/- 5% limit (intended; needs a fix to account for actual range) is problematic. Perhaps the limit issue could be addressed with yet another parameter...
2) in-air rearming could be prevented if the RPY inputs are non-zero

Reply:
I've tagged this as a dev call topic, so we can hash through it there but my rough thoughts are:

1. I'd avoid the parameter until we find out it's necessary. I suspect that the deadzone will cover the common cases, and the people who aren't will either not be within the 5% error band, or would actually benefit from redoing their RC calibration.

2. Only if you are pulling on the sticks. It'd be easy to neutralize them again, but I'm not sure it's a given that we want to bypass this normally.

Reply:
@WickedShell I asked Tridge about if its good enough for rudder arming....but he said throttle does not need be zero for rudder arming either....I am not good enough code sleuth to determine what it really is

Reply:
Yeah, it's a bit more debatable if you are actually doing throttle arming I guess, but there is a reason that method is available to the check. If they aren't trying to arm via the rudder then you shouldn't accept bad input...

Reply:
For rudder arming throttle must be zero (within the deadzone, which defaults to 30 usec)
That would be +/- 3% for a PWM range of 1000 usec.
The current test adds an additional +/- 5% beyond the deadzone, which is probably too much.

Reply:
So if this already doesn't work for rudder arming, then this only applies/helps for GCS based arming for example on plane right? In which case is there any value to the 5% range? That implies a general error with your RC setup....